### PR TITLE
Fix warnings and open files

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Especially Python 3.6 complains about unclosed files.
And PyPy can have real problems with it.  This PR *might* fix the unstable PyPy Travis job.
Note that I have fixed an unclosed file in the `example.tar` package as well.

### CODE REVIEW

File: zest/releaser/utils.py [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 import sys
 import textwrap
+
 try:
     from tokenize import detect_encoding
 except ImportError:
     detect_encoding = None
 try:
     import chardet
+
```
Comment (reinout) [inline zest/releaser/utils.py]: This newline looks strange to me.
Does pep8 want it?

**Comment by reinout** [comment]
That's a huge amount of work!

File: zest/releaser/pypi.py [hunk 2]
```diff
@@ -201,8 +237,8 @@ def distutils_servers(self):
         server from the list.
         """
         try:
-            index_servers = self.config.get(
-                'distutils', 'index-servers').split()
+            index_servers = self._get_text(
+                'distutils', 'index-servers', default='').split()
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply zest/releaser/pypi.py]: No. We split the result in the same line, so you get `''.split()`, which is the same as `[]`.

[the remark this answers is not in the document]
Comment (reinout) [reply zest/releaser/pypi.py]: Ah! I overlooked that.

File: zest/releaser/pypi.py [hunk 3]
```diff
@@ -289,7 +326,8 @@ def create_wheel(self):
             # If the wheel package is not available, we obviously
             # cannot create wheels.
             return False
-        create_setting = self._get_boolean('zest.releaser', 'create-wheel', None)
+        create_setting = self._get_boolean(
+            'zest.releaser', 'create-wheel', None)
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply zest/releaser/pypi.py]: The only change here is in whitespace. I guess the line was too wide. The code is the same.
And it is correct when you look at the next lines. When we get an explicit `True` or `False`, then we return it immediately, but when we get `None` back, then as fallback we look for the boolean `[bdist_wheel] universal` value and return that.

File: zest/releaser/tests/addchangelogentry.txt [hunk 4]
```diff
@@ -70,7 +71,8 @@ dashed.  Add multiple lines in one go.
     Checking data dict
     Question: OK to commit this (Y/n)?
     Our reply: <ENTER>
-    >>> print(open('CHANGES.txt').read())
+    >>> with open('CHANGES.txt') as f:
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply zest/releaser/tests/addchangelogentry.txt]: Apparently, yes. I knew that this was better for PyPy. But with Python 3.6 you also get warnings about this. See https://travis-ci.org/zestsoftware/zest.releaser/jobs/349542938, for example this one:

```
/home/travis/build/zestsoftware/zest.releaser/zest/releaser/tests/addchangelogentry.txt:1: 
ResourceWarning: unclosed file <_io.TextIOWrapper name='CHANGES.txt' mode='r' encoding='UTF-8'>
```

This PR also fixes other warnings that you can see there:

```
/opt/python/3.6.3/lib/python3.6/subprocess.py:766: 
ResourceWarning: subprocess 4966 is still running
```

A few remain, but those are internal warnings in `setuptools`, `docutils` and `z3c.testsetup`.

File: zest/releaser/utils.py (hunk 1 above)
Comment (mauritsvanrees) [reply zest/releaser/utils.py]: I think I ran `autopep8` over it and maybe that did it. But when I remove the line and run it again, it does not add it.
Looks strange indeed. I will remove it.
I could do that after merge too, maybe with some automated import sorting with `isort`.

Comment (mauritsvanrees) [reply zest/releaser/utils.py]: O well, I will do it on this PR. I need to resolve a conflict anyway before this can be merged.

**Pull request merged**